data/data.yaml

```
@@ -1270,3 +1270,162 @@
+    Пример: (x & y), где x и y - пропозициональные переменные, & - логическое «и»
+    2. Множество правильных алгебраических выражений (то есть выражений, определяемых следующим образом: любая алгебраическая 
+    переменная является правильным алгебраическим выражением. Если A и B - правильные алгебраические выражения, и рассматриваем 
+    в качестве алгебраической операции +, то (A + B) является правильным алгебраическим выражением, аналогично правильное алгебраическое 
+    выражение определяется и для других операций.
+    Пример: 12x + 3y
+  author: Утебаева Милена
+
+- question: >
+    Чем VPA (Visibly Pushdown Automata) отличается от PDA (Pushdown Automata)?
+    Сравнение VPA (Visibly Pushdown Automata) и PDA (Pushdown Automata)
+  answer: > 
+    Для построения VPA (Visibly Pushdown Automata) входной алфавит разбивается на три непересекающихся части: алфавит символов вызова, 
+    алфавит внутренних символов, алфавит символов возврата. При каждом считывании нового символа обрабатываемой строки, происходит 
+    изменение стека в зависимости от того, к какому подалфавиту он принадлежит: если прочитанный символ принадлежит алфавиту вызова,
+    то происходит закладывание на стек, если прочитанный символ принадлежит алфавиту внутренних символов, то стек не меняется, 
+    если прочитанный символ принадлежит алфавиту возврата, то со стека происходит снятие верхнего элемента.
+    В PDA (Pushdown Automata), в отличие от VPA, со стеком можно совершать действия не зависимо от прочитанного символа обрабатываемой строки. Также
+    PDA также отличается функцией перехода: функция перехода в PDA позволяет переходить по пустому символу, но при этом изменять стек. 
+    Пример: язык чётных палиндромов не распознаётся VPA, однако он распознаётся PDA
+  author: Утебаева Милена
+
+- question: >
+    Формулировка леммы Ардена (Arden's rule).
+    Уравнения с регулярными выражениями
+  answer: > 
+    Лемма Ардена (Arden's rule): пусть X, Y, Z - регулярные выражения, X = YX | Z, где Z не зависит от X, пустое слово не принадлежит Y,
+    Тогда имеет место следующее равенство для языка L задаваемого регулярным выражением X: L(X) = Y^(*)Z, где оператор * является 
+    итерацией Клини (Kleene star).
+    Лемма Ардена позволяет решать системы уравнений в регулярных выражениях при помощи нетерминального исключения и подстановки, 
+    однако важно требованием является отсутствие цепных правил в грамматике
+  author: Утебаева Милена
+
+- question: >
+    Определение переключающегося автомата (Alternating Finite Automaton).
+    Автомат имеющий множество конъюнктивных состояний
+  answer: >
+    Переключающийся автомат A (Alternating Finite Automaton) задаётся шестью компонентами: A = (Σ, Q, Q_∀, F, q_0, δ), где Σ - алфавит,
+    Q - множество состояний автомата, Q_∀ - множество конъюнктивных состояний, F - множество конечных состояний,
+    q_0 - стартовое состояние, δ: (Σ ∪ e) × Q -> Q, то есть, область определения функции перехода - это алфавит, 
```

```suggestion
    q_0 - стартовое состояние, δ: (Σ ∪ ε) × Q -> Q, то есть, область определения функции перехода - это алфавит, 
```

пустое слово вроде везде как ε писали, го тут тогда тоже

---

ок